core/src/main/java/org/apache/iceberg/MetadataColumns.java

```diff
@@ -49,26 +54,40 @@ private MetadataColumns() {
   public static final String DELETE_FILE_ROW_DOC = "Deleted row values";
 
   private static final Map<String, NestedField> META_COLUMNS = ImmutableMap.of(
+      SPEC.name(), SPEC,
       FILE_PATH.name(), FILE_PATH,
       ROW_POSITION.name(), ROW_POSITION,
       IS_DELETED.name(), IS_DELETED);
 
-  private static final Set<Integer> META_IDS = META_COLUMNS.values().stream().map(NestedField::fieldId)
```

Comment:
Had to change this as the partition type is not static and is handled specially.